File: app/src/main/java/com/example/learn/testme/ViewHolder/SubjecteViewHolder.java
package com.example.learn.testme.ViewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.example.learn.testme.Interface.ItemClickLIstener;
import com.example.learn.testme.R;

public class SubjecteViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

public TextView Subjecte_Nmae ;
public TextView Subjecte_Detail;

private ItemClickLIstener itemClickListener;


   public SubjecteViewHolder(View itemView){
    super(itemView);
    Subjecte_Nmae = (TextView)itemView.findViewById(R.id.subjecte_name);
    Subjecte_Detail = (TextView)itemView.findViewById(R.id.subjecte_detell);
    itemView.setOnClickListener(this);

  }

    public void setItemClickListener(ItemClickLIstener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        itemClickListener.onClick(view ,getAdapterPosition(),false);

    }
}

File: app/src/main/java/com/example/learn/testme/ViewHolder/SubjecteViewHolder.kt
package com.example.learn.testme.ViewHolder

import androidx.recyclerview.widget.RecyclerView.ViewHolder
import android.view.View
import android.widget.TextView
import com.example.learn.testme.Interface.ItemClickLIstener
import com.example.learn.testme.R

class SubjecteViewHolder(itemView: View) : ViewHolder(itemView), View.OnClickListener {
    var Subjecte_Nmae: TextView
    var Subjecte_Detail: TextView
    private var itemClickListener: ItemClickLIstener? = null
    fun setItemClickListener(itemClickListener: ItemClickLIstener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(view: View) {
        itemClickListener!!.onClick(view, adapterPosition, false)
    }

    init {
        Subjecte_Nmae = itemView.findViewById<View>(R.id.subjecte_name) as TextView
        Subjecte_Detail = itemView.findViewById<View>(R.id.subjecte_detell) as TextView
        itemView.setOnClickListener(this)
    }
}